### DISCUSSION TITLE
Deleting a temporary private fork of a repository after addressing a security advisory

### DISCUSSION THREAD

**Discussion by raisedadead** [community/community#23820] [New to GitHub] [answered]
<p>We have recently addressed a security advisory and in the process we created a temporary fork of a repository. How do we proceed to deleteing the fork? I do not see a settings page or option for the same as I would normally see as a organisation owner.</p>
<p>I could not find any documentation on this as well.</p>

**Comment by logankilpatrick** [comment]
<p>Hey! If you want to delete a private fork, you just need to follow the normal repository deleting workflow. Repo- Setting- Delete this repo- Type in the name of the repo. </p>
<p>I just tested this on one of my private repos so you should be good. </p>

**Comment by raisedadead** [comment]
<p>Hey <a class="mention" href="/logankilpatrick">@logankilpatrick</a> !</p>
<p>Thanks for your reply, however as I mentioned in my OP this is not a normal private repository.</p>
<p>We created the private repo using the security advisory feature, as explained <a href="https://help.github.com/en/github/managing-security-vulnerabilities/collaborating-in-a-temporary-private-fork-to-resolve-a-security-vulnerability" rel="nofollow noopener">here</a>.</p>
<p>This private fork was used to collaborate, and we do not see the settings option for the same. And hence did not see a way to delete this.</p>
<p>Here is a screenshot for reference:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/e80db0b99120578a8eaad8808c7be3d4422ccf96.png" data-download-href="/uploads/short-url/x6Q6tROe8RScuHML33J6hpOO9fw.png?dl=1" title="1.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/1884376/181079427-470cbd42-449e-473e-a196-03cc4393b83b.png" alt="1.png" data-base62-sha1="x6Q6tROe8RScuHML33J6hpOO9fw" width="690" height="372" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/e80db0b99120578a8eaad8808c7be3d4422ccf96_2_690x372.png, https://aws1.discourse-cdn.com/github/optimized/1X/e80db0b99120578a8eaad8808c7be3d4422ccf96_2_1035x558.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/e80db0b99120578a8eaad8808c7be3d4422ccf96.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/e80db0b99120578a8eaad8808c7be3d4422ccf96_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">1.png</span><span class="informations">1066×576 34.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>The documentation on this is also not available.</p>
<p>Thanks.</p>

**Comment by logankilpatrick** [comment]
<p>Based on my understanding, it’s not possible. Your best bet is to contact GitHub Support to manually delete that fork. </p>
<p>Though, they may have made it this way by design. I agree that this should be addressed in the docs. Maybe the word “temporary” fork implies it will auto-delete in the future but that info is not public at the moment. </p>

**Comment by raisedadead** [comment]
<p>Thanks again.</p>
<p>Yes, that is my understanding as well. And since this feature is relatively new, I am going to give them the benifit they haven’t yet considered addtional features and options like this.</p>
<p>Anyways, the docs should have something. I am unsure whom to contact about this, I will probably send an email as a feature request or something to GitHub’s support.</p>

**Comment by logankilpatrick** [comment]
<p>Hopefully they improve this. I suggest emailing GitHub support.</p>

**Comment by raisedadead** [comment]
<p>Update 1:</p>
<p>You can not delete a temporary fork of a repo, which you created as a part of an open security advisory, if you close the advisory without making the security advisory public. This will include cases, where to chose not to disclose the advisory, or the advisory was not valid and could be closed. Regardless, if you created any temporary forks as a part of it, they will only be delete after you make the advisory public.</p>
<p>There is no documenation clarifiing this.</p>
<p>Once you do make the advisory public and close it, you should be able to delete any provate forks created.</p>
<p>Update 2:</p>
<p>GitHub support has reached out to me, I am awaitng for a reponse after clarifying the above details.</p>

**Comment by raisedadead** [comment]
<p>Update:</p>
<p>The Support team got back to me, acknowledging this to be a gap in the docs. They have forwarded the feedback to the engineering team. No commitments on a fix though.</p>
<p>Hope this helps someone with a similar scenario.</p>

**Reply by jakubgs** [reply]
How about now?

**Comment by santrancisco** [comment]
Oh dear - we are now at 2023 and I don't think this is fixed yet? @github-staff  - can you confirm this is the case or I missed something? (I don't see settings or other tabs allowing me to delete the private fork)

**Reply by Trisraki** [reply]
Any updates?

 @github-staff ???

**Reply by glye** [reply]
Also, I keep getting Dependabot alerts from a temp private repo where the advisory has been closed (not published). I have tried setting it to "ignored" in notification settings, hoping that will work. But we definitely ought to be able to delete these repos!

**Reply by maxime-rainville** [reply]
Still a problem. I tried deleting the repo via an API request but I got a "403 Must have admin rights to Repository" response back.

**Comment by Guru-98** [comment]
Any updates on this? @github-staff

**Comment by ScriptAutomate** [comment]
This same problem came up for me in a separate, partially-related discussion:

- [GitHub Discussion: Extend APIs related to GitHub Security Advisories and their Temporary Private Forks](https://github.com/orgs/community/discussions/60800)

**Comment by raisedadead** [comment, marked as answer]
It seems they have finally gotten around to addressing it.

<https://github.blog/changelog/2023-09-29-temporary-private-forks-can-now-be-deleted-from-repository-advisories>

**Reply by bashonly** [reply]
I still can't delete the temporary private fork from the repository advisory

**Reply by minrk** [reply]
I just ran into this for an old closed advisory. There is no button to delete the fork for a closed advisory. When I reopen the advisory, the delete button appears, then I can close the advisory again.

**Reply by frankie567** [reply]
I confirm that if the advisory is closed, you don't have the button to delete it. You need to reopen the advisory, delete the fork and close the advisory again. That's super annoying.

**Reply by raisedadead** [reply]
I would just reachout to GitHub support if I were you. At this point it would be considered a bug.
